scrape/target.go

```diff
@@ -366,15 +366,15 @@ type bucketLimitAppender struct {
 func (app *bucketLimitAppender) AppendHistogram(ref storage.SeriesRef, lset labels.Labels, t int64, h *histogram.Histogram, fh *histogram.FloatHistogram) (storage.SeriesRef, error) {
 	if h != nil {
 		for len(h.PositiveBuckets)+len(h.NegativeBuckets) > app.limit {
-			if h.Schema == -4 {
+			if h.Schema == nativeHistogramMinSchema || h.Schema > nativeHistogramMaxSchema {
```

Comment:
Let's do <= on left and also we need to rename it to nativeExponentialMinSchema, nativeExponentialHistogramMaxSchema on `main`, because the name is wrong now.